### PR TITLE
TYP/CLN: remove no-longer-needed overrides

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/series.py [hunk 1]
```diff
@@ -4352,43 +4352,6 @@ def aggregate(self, func=None, axis: Axis = 0, *args, **kwargs):
...
-        bool_only: bool | None = ...,
-        skipna: bool = ...,
-        level: Level,
-        **kwargs,
-    ) -> Series | bool:
-        ...
-
-    # error: Missing return statement
-    @doc(NDFrame.any, **_shared_doc_kwargs)
-    def any(  # type: ignore[empty-body]
```
Comment (twoertwein) [inline pandas/core/series.py:4382]: `NDFrame.any` doesn't seem to have a doc-string.

Comment (twoertwein) [reply pandas/core/series.py:4382]: (if possible, it would be great to add overloads to `NDFrame.any` to avoid returning `bool | Series | DataFrame`)

Comment (jbrockmendel) [reply pandas/core/series.py:4382]: it gets re-defined with a docstring in _add_numeric_operations (which is a pattern im leaning towards getting rid of)

File: pandas/core/frame.py [hunk 2]
```diff
@@ -11795,232 +11795,6 @@ def values(self) -> np.ndarray:
...
-        self,
-        *,
-        axis: None | Axis = None,
-        inplace: bool = False,
-        limit: None | int = None,
-        downcast=None,
-    ) -> DataFrame | None:
-        return super().bfill(axis=axis, inplace=inplace, limit=limit, downcast=downcast)
-
-    def clip(
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/core/frame.py:11884]: looks like lower and upper cna be listlike, so the DataFrame annotation is inaccurate

**Review by topper-123 — APPROVED** [review]
LGTM. 500 removed lines is just awesome.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**